src/app/charts/extra-params-components/percentile.component.ts

```diff
@@ -0,0 +1,57 @@
+import { AfterViewInit, Component, EventEmitter, Input, Output, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+import { Indicator } from '../../models/indicator.model';
+import { PercentileIndicatorQueryParams } from '../../models/percentile-indicator-query-params.model';
+
+import * as _ from 'lodash';
+
+/*
+ * Percentile params component
+ * Uni-field form to allow user to specify the percentile params
+ */
+@Component({
+  selector: 'ccl-percentile-parameters',
+  templateUrl: './percentile.component.html'
+})
+export class PercentileComponent implements AfterViewInit, OnInit {
+
+    @Input() indicator: Indicator;
+    @Input() extraParams: PercentileIndicatorQueryParams;
+    @Output() percentileParamSelected = new EventEmitter<PercentileIndicatorQueryParams>();
+
+    percentileForm: FormGroup;
+
+    // default form values
+    private defaultPercentile = 50;
+
+    constructor(private formBuilder: FormBuilder) {}
+
+    ngOnInit() {
+        // must create form on init instead of constructor to capture @Input values
+        this.createForm();
+    }
+
+    ngAfterViewInit() {
+        // Since valueChanges triggers initially before parent is ready, wait until
+        // parent is ready here and trigger it to draw chart with extra parameters.
+        this.percentileParamSelected.emit({
+            'percentile': this.percentileForm.controls.percentileCtl.value
+        });
+    }
+
+    createForm() {
+        this.percentileForm = this.formBuilder.group({
+            percentileCtl: [this.extraParams.percentile || this.defaultPercentile, Validators.required]
+        });
+
+        this.percentileForm.valueChanges.debounceTime(700).subscribe(form => {
+            // only accept percentiles [1, 100] as integers
+            const pctl = form.percentileCtl;
+            if (pctl > 100 || pctl < 1) { return; }
+            this.percentileParamSelected.emit({
+                'percentile': Math.round(pctl)
```

Comment:
Using [ng-pattern](https://docs.angularjs.org/api/ng/directive/ngPattern) would be better, to enforce input restrictions rather than implicitly rounding what the user enters. `ng-pattern="/^-?[0-9][^\.]*$/"` should do the trick.

Although there are restrictions on the input now, there's no feedback to the user to indicate that the input is invalid; the chart simply fails to reload. It would be good to maybe highlight the field red or give some other error feedback.